relay-kafka/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+//! Kafka-related functionality.
+//!
+//! By default, this library only provides the support for `relay` configuration. With `producer`
+//! feature enabled, the [`KafkaClient`] is provided and can be used to send messages to the Kafka
+//! broker.
+//!
+//! # Usage
+//!
+//! ```compile_fail
```

Comment:
It's unfortunate that we cannot even compile this because of missing features. ~Maybe we move the documentation of the producer part into the `producer` module so it can compile?~ (that won't work since the module is not public, just its contents)

Regardless, we should at least insert the correct imports here.